src/lib/transaction_protocol_state/transaction_protocol_state.mli

```diff
@@ -0,0 +1,51 @@
+open Core
```

Comment:
Can you add some doc-comments to explain the purpose and role of a `Transaction_protocol_state.t`? Perhaps as well some details around the `Block_data.t` as well. For instance, it's unclear to me (immediately, at least) why the `Block_data.t` is an optional state body hash.